---
title: "Protein Tau's Role in Gene Expression"
author: "Group 25:  Ana  Moral (s232119),  Jacqueline Printz (s194377),  Jenni Kinnunen (s204697), Jo√£o Prazeres (s243036), William Gunns (s242051)"
format:
  revealjs:
    slide-number: true
    embed-resources: true 
    theme: night
    smaller: true 
    scroll: true
---

### 1. Introduction - Protein Tau

::: {style="display: flex; align-items: flex-start; justify-content: space-between;"}
::: {style="width: 55%; font-size: 22px; text-align: justify;"}
-   **Function**: Microtubule protein essential for cytoskeletal stability and neuronal transport.\
    \

-   Supports healthy neuronal functions.\
    \

-   Destabilization linked to neuronal dysfunction, and **Alzheimer's Disease**.\
    \

-   Previous studies concluded that Tau destabilization led to an alteration in the expression of glutamatergic genes.\

[**Experimental Objective:**]{style="width: 45%; font-size: 120%; text-align: justify;"}\
\
[Is the overexpression of Tau associated to gene expression alterations?]{style="width: 45%; font-size: 120%; text-align: justify;"}
:::

::: {style="width: 45%; text-align: right;"}
<img src="images/Tau_Protein.png" alt="Tau Protein Diagram" style="width: 100%; height: 600px;"/>
:::
:::

------------------------------------------------------------------------

### 2. Experimental Setup

::: {style="display: flex; align-items: flex-start; justify-content: space-between;"}
::: {style="width: 45%; font-size: 22px; text-align: justify;"}
**Differential gene expression analysis of RNA-seq data performed on**:

-   **Control**: 3 samples of SH-SY5Y cells with overexpression of a control vector.\
-   **Experimental Condition**: 3 samples of SH-SY5Y cells with overexpression of Tau 0N4R isoform.

**RNA-seq data was reported on 3 xls sheets:**

-   **Read Counts**.\
-   **RPM** (Reads Per Million).\
-   **RPKM** (Reads Per Kilobase Million).\

The 3 sheets were joined into one large tibble data frame.
:::

::: {style="width: 50%; text-align: right;"}
```{r, echo = FALSE, message=FALSE}
library("tidyverse")
reads_df <- read_tsv("../data/01_data_load_reads.tsv.gz")
reads_df
```
:::
:::

------------------------------------------------------------------------

### 3. Data Wrangling

::: columns
::: {.column width="50%" style="font-size: 26px;"}
**First the data was prepared and made clean by:**\
\
1. Joining three RNA sequencing data sheets into one.\
\
2. Renaming columns and naming unamed columns.\
\
3. Removing unecessary and invalid observations. \
\
4. Descreption data (gene_Ensmebl, gene_ID, gene_decription) was saved in a metadata file.\

:::
::: {.column width="50%" style="font-size: 26px;"}

**Tidyverse functions used:**\
\
\
**full_join:** to merge three sheets into one dataframe\
\
**mutate:** add new columns to dataset\
\
\
**select:** to subset and/or remove relevant columns\
\
**filter:** to subset and/or remove relevant rows
:::
:::

------------------------------------------------------------------------

### 4. Data Augment

::: columns ::: {.column width="40%" style="font-size: small;"} ::: incremental Normalized Data

-   Log transformation applied to selected columns
-   Small value (0.0001) added to avoid zeros in data

Calculated Mean - Mean values for control and tau groups were calculated across replicates

Fold Change - Calculated fold change between control and tau groups for each measure - Fold change values used to filter significant genes

Filtered Data - Genes with significant fold change (\>1 or \<-1) retained - Replicates averaged for non-significant genes

Final Data -Data stored in three separate files for analysis ::: ::: ::: {.column width="60%" style="font-size: small;"}{fig-align="right" width="300"} :::

------------------------------------------------------------------------

### 5. Data Description part 1

::: columns
::: {.column width="40%" style="font-size: 22px;"}
::: incremental
All data

-   522,648 observations, 3 attributes

-   29,036 genes

-   18 experiments, each of them have 3 replicates
:::
:::

::: {.column width="60%" style="font-size: 22px;"}
::: incremental
Filtered data

-   110,958 observations

-   19,379 genes
:::
:::
:::

------------------------------------------------------------------------

### 6. Data Description part 2
<img src="images/describe plot.png" style="width: 80%;"/>

- Genes with greater natural expression appear to be more effected by the increase in Tau protein.\
- The x axis includes negative values for the RPM and RPKM plots, this is due to the log transform of values less than 1. The same effect is not present on the reads graph due to integer values.


------------------------------------------------------------------------

### 7. Analysis PCA

::: columns ::: {.column width="40%" style="font-size: small;"} ::: incremental Objective

-   Confirm that RPM, RPKM, and reads yield similar results
-   Verify differences between control and tau experiments

Approach - 3 PCAs performed separately for RPM, RPKM, and reads data - 1 final PCA conducted on combined data

Results - Plots of individual PCAs show how each data type clusters - Final PCA confirms global differences between control and tau groups ::: ::: ::: {.column width="60%" style="font-size: small;"} {fig-align="right" width="300"} ::: ------------------------------------------------------------------------

### 8. Analysis PCA

Maybe for plots

------------------------------------------------------------------------

### 9. Gene Set Enrichment Analysis (GSEA)

Computational method to determine if a set of genes shows statistically significant differences in control and Tau over expressing conditions.


::: {.left}
-   Plot genes

::: {style="width: 45%; text-align: left;"}
<img src="images/gene_plot.png" alt="Over and under expressed genes" style="width: 100%; height: 600px;"/>

::: {.right}
-   Plot pathways

::: {style="width: 45%; text-align: right;"}
<img src="images/gsea_plot.png" alt="Over and under expressed pathways" style="width: 100%; height: 600px;"/>
------------------------------------------------------------------------

### 10. Discussion based on the GSEA/conclusion

Which genes were overexpressed? Does it make sense with the literature?

gamma - cytokine, regulation of immune response, trigger jak stat
alpha - cytokine, pathogen recognition response, trigger jak stat
jak stat - immunity, cell division,   --promote neuroinflammation in neurodegenerative disorders like Alzheimer's by initiation innate immunity, adaptive immune mechanisms, and neuroinflammatory response.(https://pubmed.ncbi.nlm.nih.gov/36614305/#:~:text=The%20JAK%2FSTAT%20signaling%20pathway%20is%20one%20of%20the%20critical,and%20finally%2C%20constraining%20neuroinflammatory%20response.)


Challenges

-   \[Challenge 1\]

Data handling and cleaninig: setting arbitrary boundaries wothout losing too much data or leaving in unnecessary data. 

-   \[Challenge 2\]

Limitations

-   \[Limitation 1\]
-   \[Limitation 2\]
