---
title: "R for Bio Data Science"
author: "Group 25"
format:
  revealjs:
    slide-number: true
    embed-resources: true 
    theme: dark
---

#### 1. Introduction

::: {style="display: flex; align-items: flex-start; justify-content: space-between;"}
::: {style="width: 45%; font-size: 50%; text-align: justify;"}
### Protein Tau

-   **Function**: Microtubule protein essential for cytoskeletal stability and neuronal transport.\
-   **Key Role**: Supports healthy neuronal functions.\
-   **Pathology**:
    -   Destabilization linked to neuronal dysfunction, death, and **Alzheimer's Disease**.\
    -   Aggregates contribute to neurodegeneration.

------------------------------------------------------------------------

### Experimental Objective

-   Previous studies concluded that Tau destabilization led to an alteration in the expression of glutamatergic genes.

Experiment Objective:\
\
[**Is the overexpression of Tau associated to gene expression alterations?**]{style="width: 45%; font-size: 120%; text-align: justify;"}
:::

::: {style="width: 55%; text-align: right;"}
<img src="images/Tau_Protein.png" alt="Tau Protein Diagram" style="width: 100%; height: 600px;"/>
:::
:::

------------------------------------------------------------------------

#### 2. Experimental Setup

::: {style="display: flex; align-items: flex-start; justify-content: space-between;"}
::: {style="width: 45%; font-size: 50%; text-align: justify;"}
### Overall Design

**Differential gene expression analysis of RNA-seq data performed on**:

-   **Control**: 3 samples of SH-SY5Y cells with overexpression of a control vector.\
-   **Experimental Condition**: 3 samples of SH-SY5Y cells with overexpression of Tau 0N4R isoform.

------------------------------------------------------------------------

### Results

RNA-seq data was reported on 3 xls sheets:

-   **Read Counts** sheet: Total number of reads for each gene.\
-   **RPM** (Reads Per Million) sheet: Read Counts are divided by a million factor (total number of reads divided by 1 million).\
-   **RPKM** (Reads Per Kilobase Million) sheet: Corresponds to RPM divided by the length of each gene.

The 3 sheets were joined into one large tibble data frame.
:::

::: {style="width: 50%; text-align: right;"}
```{r, echo = FALSE, message=FALSE}
library("tidyverse")
reads_df <- read_tsv("../data/01_data_load_reads.tsv.gz")
reads_df
```
:::
:::

------------------------------------------------------------------------

#### 3. Data Wrangling

::: columns
::: {.column width="40%" style="font-size: 50%;"}
::: incremental

First the data was prepared and made clean by,
1. by joining all dataframes into one
2. renaming columns
3. removing unecessary and invalid observations (including descriptions and rows of data that are all zero)
:::
Some of the key functions that were used include full_join, mutate, rename, select)
Following this, each row corresponds to an observation, each column corresponds to a variable and each cell is a value. Could insert picture of 'clean' data - lecture style.
:::

::: {.column width="60%" style="font-size: small;"}
::: incremental
:::
The data was then log transformed to enable a deeper analysis of observations with small margins of difference.
Average across trials of experiment (and variance) THINK THIS BIT IS FOR AUGMENT
:::
:::

------------------------------------------------------------------------

#### 4. Data Augment 


::: columns
::: {.column width="40%" style="font-size: small;"}
::: incremental
Statistics data frame

-   Variance for the 3 replicates for each attributes

Filtered data

-   Genes with mean variance \> 1 were filtered out

-   For the rest, average of replicates was kept

    -   Data frame 3 times smaller
:::
:::

::: {.column width="60%" style="font-size: small;"}
![](images/data.png){fig-align="right" width="300"}
:::
:::

------------------------------------------------------------------------

#### 5. Data Description part 1

::: columns
::: {.column width="40%" style="font-size: small;"}
::: incremental
All data

-   522,648 observations, 3 attributes

-   29,036 genes

-   18 experiments, each of them have 3 replicates
:::
:::

::: {.column width="60%" style="font-size: small;"}
::: incremental
Filtered data

-   110,958 observations

-   19,379 genes
:::
:::
:::

------------------------------------------------------------------------

#### 6. Data Description part 2

Insert picture of plot
Similar description to that in the report


------------------------------------------------------------------------

#### 7. Analysis  PCA

The objective was to confirm that RPM, RPKM and reads give the same results, as well as confirm that there is in fact a difference when comparing the control and the tau experiment.

We performed 3 PCAs (the first 3 plots) on each type of results, and 1 final PCA for all. 

------------------------------------------------------------------------

#### 8. Analysis  PCA
Maybe for plots

------------------------------------------------------------------------

## 9. Gene Set Enrichment Analysis
Text


------------------------------------------------------------------------

#### 10. Discussion based on the GSEA/conclusion  

Which genes were overexpressed? Does it make sense with the literature?

Challenges

-   \[Challenge 1\]
-   \[Challenge 2\]

Limitations

-   \[Limitation 1\]
-   \[Limitation 2\]


