---
title: "R for Bio Data Science"
author: "Group 25"
format:
  revealjs: 
    theme: dark
---

#### 1. Introduction

<div style="display: flex; align-items: flex-start; justify-content: space-between;">

<div style="width: 45%; font-size: 55%; text-align: justify;">
  
### Protein Tau

- **Function**: Microtubule protein essential for cytoskeletal stability and neuronal transport.  
- **Key Role**: Supports healthy neuronal functions.  
- **Pathology**:  
    - Destabilization linked to neuronal dysfunction, death, and **Alzheimer's Disease**.  
    - Aggregates contribute to neurodegeneration.  

---

### Experimental Objective

- Previous studies concluded that Tau destabilization led to an alteration in the expression of glutamatergic genes.   

What we want to conclude:  
**Is the overexpression of Tau associated to gene expression alterations?**

</div>

<div style="width: 55%; text-align: right;">
  <img src="images/Tau_Protein.png" alt="Tau Protein Diagram" style="width: 100%; height: 600px;">
</div>

</div>
------------------------------------------------------------------------

#### 1.2 Experimental Setup

<div style="display: flex; align-items: flex-start; justify-content: space-between;">

<div style="width: 45%; font-size: 50%; text-align: justify;">

### Overall Design

**Differential gene expression analysis of RNA-seq data performed on**:  

- **Control**: 3 samples of SH-SY5Y cells with overexpression of a control vector.  
- **Experimental Condition**: 3 samples of SH-SY5Y cells with overexpression of Tau 0N4R isoform.  

---

### Results

RNA-seq data was reported on 3 xls sheets:  

- **Read Counts** sheet: Total number of reads for each gene.  
- **RPM** (Reads Per Million) sheet: Read Counts are divided by a million factor (total number of reads divided by 1 million).  
- **RPKM** (Reads Per Kilobase Million) sheet: Corresponds to RPM divided by the length of each gene.    

The 3 sheets were joined into one large tibble data frame.

</div>

<div style="width: 50%; text-align: right;">
```{r, echo = FALSE, message=FALSE}
library("tidyverse")
reads_df <- read_tsv("../data/01_data_load_reads.tsv.gz")
reads_df
```

  
</div>

</div>


------------------------------------------------------------------------

#### 2. Materials and Methods: Augment part 1

::: columns
::: {.column width="40%" style="font-size: small;"}
::: incremental
text text
:::
:::

::: {.column width="60%" style="font-size: small;"}
::: incremental
text
:::
:::
:::

------------------------------------------------------------------------

#### 3. Materials and Methods: Augment part 2

::: columns
::: {.column width="40%" style="font-size: small;"}
::: incremental
Statistics data frame

-   Variance for the 3 replicates for each attributes

Filtered data

-   Genes with mean variance \> 1 were filtered out

-   For the rest, average of replicates was kept

    -   Data frame 3 times smaller
:::
:::

::: {.column width="60%" style="font-size: small;"}
![](images/data.png){fig-align="right" width="300"}
:::
:::

------------------------------------------------------------------------

#### 4. Materials and Methods: Describe part 1

::: columns
::: {.column width="40%" style="font-size: small;"}
::: incremental
All data

-   522,648 observations, 3 attributes

-   29,036 genes

-   18 experiments, each of them have 3 replicates
:::
:::

::: {.column width="60%" style="font-size: small;"}
::: incremental
Filtered data

-   110,958 observations

-   19,379 genes
:::
:::
:::

------------------------------------------------------------------------

#### 5. Materials and methods: Describe part 2

Plots

------------------------------------------------------------------------

#### 6. Materials and methods: Analysis, PCA

------------------------------------------------------------------------

## 7. Materials and methods: Analysis, GSEA

------------------------------------------------------------------------

#### 8. Results

Key Findings\
- \[Finding 1\] - \[Finding 2\] - \[Finding 3\]

------------------------------------------------------------------------

#### 9. Discussion

Challenges

-   \[Challenge 1\]
-   \[Challenge 2\]

Limitations

-   \[Limitation 1\]
-   \[Limitation 2\]

------------------------------------------------------------------------

#### 10. Conclusion

Summary

-   Key takeaways from the analysis

Future Work

-   \[Plan for next steps\]
