---
title: "Describe data"
format: html
editor: visual
---

# Load data & libraries

```{r}
#| message: false
library("tidyverse")
library("dplyr")
library(patchwork)
```

From the augment script, we will use 2 of the files.

1\) All data in long format

2\) Filtered data, where genes with too high mean SD between replicates have been filtered out, and the average of the replicates is used for the rest.

```{r}
data_dir <- "../data"
data_file_long <- "/03_data_load.tsv.gz"
filtered_data_file <- "/03_data_filtered_load.tsv.gz"

all_data_long <- read_tsv(str_c(data_dir, data_file_long))
filtered_data <- read_tsv(str_c(data_dir, filtered_data_file))
```

# Size of the data and number of genes

After tidying the data, we now have 522648 observations and 3 attributes.

```{r}

dim(all_data_long)
```

There are 29036 different genes

```{r}
unique_genes <- all_data_long |>
  select(gene_ensembl) |>
  distinct()

dim(unique_genes)
```

Let's look at the experiments. There are 18 per gene.

```{r}
all_data_long |>
  group_by(gene_ensembl) |>
  summarise(row_count = n())
```

# Attributes

For each gene ID, there are

-   mean_reads: raw count of sequencing reads mapped to a gene

-   mean_rpm: normalized value of mean_reads

-   mean_rpkm: normalized value of mean_rpm, by considering length of gene (in kilobases)

-   gene expression value: how much the gene is expressed (i.e. how actively it's producing its corresponding RNA)

# Gene expression values

Let's find the range of the gene expression values. The log transformation performed in the augment script explains the negative values.

```{r}
gene_expression_summary <- all_data_long |>
  #group_by(gene_ID) |>
  summarize(
    mean_value = mean(gene_expression),
    min_value = min(gene_expression),
    max_value = max(gene_expression),
    count = n())

gene_expression_summary
```

# Filtered data

In the filtered data, there are 19379 genes left

```{r}
unique_genes <- filtered_data |>
  select(gene_ensembl) |>
  distinct()

dim(unique_genes)
```

The filtered data has 110958 observations in total.

```{r}
dim(filtered_data)
```

# Visualizing the data

The data has been visualised below in separated plots for each experiment. Despite not accounting for different gene types, the plots depict a clear trend, with the control samples consistantly presenting greater gene expression. Furthermore, the fact that the control and tau curves follow such a similar shape indicates that is it likely that the increased tau protein has caused a linear shift in the gene expression.

```{r}
RPKMplot <- filtered_data |>
  filter(str_detect(experiment,
                    pattern = "rpkm")) |> 
  ggplot(mapping = aes(x = gene_expression,
                       fill = experiment)) + 
  geom_density(alpha = 0.5) + 
  theme_minimal() + 
  labs(title = "RPKM Data",
       y = "Density",
       x = "Mean Gene Expression")

RPMplot <- filtered_data |>
  filter(str_detect(experiment,
                    pattern = "rpm")) |> 
  ggplot(mapping = aes(x = gene_expression,
                       fill = experiment)) + 
  geom_density(alpha = 0.5) + 
  theme_minimal() + 
  labs(title = "RPM Data",
       y = "Density",
       x = "Mean Gene Expression")

readsplot <- filtered_data |>
  filter(str_detect(experiment,
                    pattern = "reads")) |> 
  ggplot(mapping = aes(x = gene_expression,
                       fill = experiment)) + 
  geom_density(alpha = 0.5) + 
  theme_minimal() + 
  labs(title = "Reads Data",
       y = "Density",
       x = "Mean Gene Expression")

# Combining the plots above each other
combined_plot <- RPKMplot / RPMplot / readsplot
combined_plot
```

```{r}
filtered_data |>
  filter(str_detect(experiment, pattern = "rpkm"))

```
